### ISSUE TITLE
Add FormKitNode findUpwards() function

### ISSUE THREAD

**Issue by justin-schroeder** [formkit/formkit#1198]
### Discussed in https://github.com/orgs/formkit/discussions/1196

<div type='discussions-op-text'>

<sup>Originally posted by **mlmoravek** February  9, 2024</sup>
## Feature: 

A `FormKitNode` element has the [`at()`](https://formkit.com/api-reference/formkit-core#at), [`find()`](https://formkit.com/api-reference/formkit-core#find) and the [`walk()`](https://formkit.com/api-reference/formkit-core#walk) functions to walk through the subtree and find a particular node. For going upwards and searching there are only the [`parent`](https://formkit.com/api-reference/formkit-core#parent) and [`root`](https://formkit.com/api-reference/formkit-core#root) properties. 
However, I'm missing a function to walk up the tree until a node matches some condition.

I would like to have some function like `findUpward()` to find a node which is located up the tree of the current node. 
</div>

**Comment by andrew-boyd** [comment]
Something like `closest()` is probably the most common naming convention for that behaviour.

**Comment by fenilli** [comment]
I agree with `closest()` but would it gives the idea that it will also go downwards, it will find a sibling, child or parent.

**Comment by sethatwood** [comment]
To @andrew-boyd's suggestion of `closest()` and @fenilli's concern about ambiguity, we can see in the MDN docs that `Element.closest()` only traverses upwards towards the DOM root:

> The closest() method of the [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) interface traverses the element and its parents (heading toward the document root) until it finds a node that matches the specified [CSS selector](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Selectors).

https://developer.mozilla.org/en-US/docs/Web/API/Element/closest 

I personally vote for adhering to a common, well-documented naming convention that already means "traverses upward" over introducing a new naming convention that is less ambiguous.
